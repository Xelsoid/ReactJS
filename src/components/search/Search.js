import React from 'react';

import './search.scss';
import PropTypes from "prop-types";

const Search = ({callback, value}) => {
  return (
    <div className="search-box__wrapper">
      <label className="search-box__search-label" htmlFor="searchFilm">
        <i aria-hidden="true" className="search-box__search-icon fa fa-search" />
      </label>
      <input
        className="search-box__search-input"
        id="searchFilm"
        placeholder="search"
        type="search"
        value={value}
        onChange={callback}
      />
    </div>
  );
};

Search.defaultProps = {
  callback: null,
  value: ''
};

Search.propTypes = {
  callback: PropTypes.func,
  value: PropTypes.string
};

export default Search;
