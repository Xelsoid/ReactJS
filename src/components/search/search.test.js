import React from 'react';
import Search from './Search';

it('renders correctly', () => {
    const component = shallow (
      <Search />
    );

    expect(component).toMatchSnapshot();
});

