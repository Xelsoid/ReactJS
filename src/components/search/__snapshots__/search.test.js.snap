// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="search-box__wrapper"
>
  <label
    className="search-box__search-label"
    htmlFor="searchFilm"
  >
    <i
      aria-hidden="true"
      className="search-box__search-icon fa fa-search"
    />
  </label>
  <input
    className="search-box__search-input"
    id="searchFilm"
    onClick={[Function]}
    placeholder="search"
    type="search"
  />
</div>
`;
